website/content/docs/concepts/node-pools.mdx

```diff
@@ -0,0 +1,366 @@
+Some examples include reverse proxies for traffic ingress, CSI plugins, and
+periodic maintenance jobs.
+
+These jobs can be isolated in their own namespace but they may have different
+scheduling requirements.
+
+Reverse proxies, and only reverse proxies, may need to run in clients that are
+exposed to public traffic, and CSI controller plugins may require clients to
+have high-privileged access to cloud resources and APIs.
+
+Other jobs, like CSI node plugins and periodic maintenance jobs, may need to
+run as `system` jobs in all clients of the cluster.
+
+Node pools can be used to achieve the isolation required by the first set of
+jobs, and the built-in `all` node pool can be used for the jobs that must run
+in every client. To keep them organized, all jobs are registered in the same
+`infra` namespace.
+
+```hcl
+job "ingress-proxy" {
+  namespace = "infra"
+  node_pool = "ingress"
+  # ...
+}
+```
+
+```hcl
+job "csi-controller" {
+  namespace = "infra"
+  node_pool = "csi-controllers"
+  # ...
+}
+```
+
+```hcl
+job "csi-nodes" {
+  namespace = "infra"
+  node_pool = "all"
+  # ...
+}
+```
+
+```hcl
+job "maintenance" {
+  type      = "batch"
+  namespace = "infra"
+  node_pool = "all"
+
+  periodic { /* ... */ }
+  # ...
+}
+```
+
+Use positive and negative constraints to fine-tune placements when targeting
+the built-in `all` node pool.
+
+```hcl
+job "maintenance-linux" {
+  type      = "batch"
+  namespace = "infra"
+  node_pool = "all"
+
+  constraint {
+    attribute = "${attr.kernel.name}"
+    value     = "linux"
+  }
+
+  constraint {
+    attribute = "${node.pool}"
+    operator  = "!="
+    value     = "ingress"
+  }
+
+  periodic { /* ... */ }
+  # ...
+}
+```
+
+With Nomad Enterprise and Node Pool Governance, the `infra` namespace can be
+configured to use a specific namespace by default and only allow the specific
+node pools required.
+
+```hcl
+namespace "infra" {
+  description = "Infrastructure jobs."
+
+  node_pool_config {
+    default = "infra"
+    allowed = ["ingress", "csi-controllers", "all"]
+  }
+}
+```
+
+### Mixed Scheduling Algorithms
+
+The different scheduling algorithms provided by Nomad are best suited for
+different types of environments and workloads.
+
+The `binpack` algorithm aims to maximize resource usage and pack as much
+workload as possible in the given set of of clients. This makes it ideal for
+cloud environments where infrastructure is billed by the hour and can be
+quickly scaled in and out. By maximizing workload density a cluster running in
+cloud instances can reduce the number of clients needed to run everything that
+is necessary.
+
+The `spread` algorithm behaves in the opposite direction, making use of every
+client available to reduce density and potential noisy neighbors and resource
+contention. This makes it ideal for environments where clients are
+pre-provisioned and scale more slowly, such as on-premise deployments.
```

Comment:
```suggestion
pre-provisioned and scale more slowly, such as on-premises deployments.
```

IMO, this is one of those things the "grammar police" get overly worked up about, but better to avoid that discussion. :grinning:

Reply:
I don't think I ever spelled out "on-premises" in full, I usually just go with "on-prem", so good to know 😄